main.ts

```diff
@@ -114,6 +118,26 @@ export default class TikzjaxPlugin extends Plugin {
 		window.CodeMirror.modeInfo = window.CodeMirror.modeInfo.filter(el => el.name != "Tikz");
 	}
 
+	// Inspired by Chris Morgan's answer in https://stackoverflow.com/questions/33631041/javascript-async-await-in-replace
+	async replaceAsync(string: string, regexp: RegExp, replacer: (...args: any[]) => Promise<string>): Promise<string> {
+		const replacements = await Promise.all(
+			Array.from(string.matchAll(regexp),
+				match => replacer(...match)));
+		let i = 0;
+		return string.replace(regexp, () => replacements[i++]);
+	}
+
+	async embedTexInput(tikzSource: string): Promise<string> {
+		return this.replaceAsync(tikzSource, /^%:input\s+(.+)$/gm, async (_, path) => {
+			const file = this.app.vault.getAbstractFileByPath(path);
+			if (file instanceof TFile) {
```

Comment:
Should we print a warning to the console if a user inputs a filepath that Obsidian can't locate?